.github/workflows/ruby.yml

```diff
@@ -18,20 +18,33 @@ jobs:
   test:
 
     runs-on: ubuntu-latest
+    services:
+      postgres:
+        image: postgres:12.3
+        env:
+          POSTGRES_PASSWORD: notinproduction
+        ports:
+          - 5432:5432
+        options: --health-cmd pg_isready --health-interval 10s --health-timeout 5s --health-retries 5
```

Comment:
Don't quite get this line. Shouldn't the timeout duration be longer than the interval duration, otherwise it will always timeout before an interval is finished? 
https://codeblog.dotsandbrackets.com/docker-health-check/

Reply:
I daresay that blog post is wrong. Definitely misleading. As per [official documentation](https://docs.docker.com/engine/reference/run/#healthcheck):

`--health-interval` is how often I run the `--health-cmd`, so I'm running `pg_isready` every `10s`. The time within which I expect a response is dictated by `--health-timeout`, so I expect a check to complete and respond within `5s` of being requested. If `5` consecutive checks fail, as set by `--health-retries`, the container's status will change to `unhealthy`.

Reply:
Further reading, I remembered seeing this (and then shamelessly copied) from [Andy Croll's newsletter](https://andycroll.com/ruby/github-actions-ci-for-rails-with-postgresql/), which I recommend subscribing to